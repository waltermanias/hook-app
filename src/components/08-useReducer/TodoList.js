import React from "react";
import PropTypes from "prop-types";
import TodoListItem from "./TodoListItem";

export const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => {
        return (
          <TodoListItem
            handleDelete={handleDelete}
            handleToggle={handleToggle}
            todo={todo}
            key={todo.id}
            index={i}
          />
        );
      })}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoList;
