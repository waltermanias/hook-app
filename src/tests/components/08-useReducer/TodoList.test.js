import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";

describe("TodoList", () => {
  it("should render the component", () => {
    const wrapper = shallow(
      <TodoList
        todos={[
          { id: 1, done: false, desc: "test-desc" },
          { id: 2, done: true, desc: "test-desc-2" },
        ]}
        handleDelete={() => {}}
        handleToggle={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("TodoListItem").length).toBe(2);

    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
