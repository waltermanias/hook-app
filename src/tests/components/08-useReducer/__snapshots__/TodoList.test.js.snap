// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render the component 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[Function]}
    handleToggle={[Function]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "test-desc",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[Function]}
    handleToggle={[Function]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "test-desc-2",
        "done": true,
        "id": 2,
      }
    }
  />
</ul>
`;
